keyboards/moon/pca9555.c

```diff
@@ -0,0 +1,78 @@
+/* Copyright 2019
```

Comment:
Who is this file actually licensed to/by/for?

Reply:
I have no idea. I found it in `keyboards/xd84`. Though I made a few adjustments to it for things that I believe are bugs.

Reply:
Ah, okay.  Then a better question, that was not caught would be at @zvecr

Reply:
I was previously told that if i did not want personal attribution, to just leave it at `Copyright 2019` without my name, however if there has been a shift on this stance then i am happy to update, ie to something QMK releated `Copyright 2019 QMK`.

As for the bugs, i diffed the files and can only see the updated comments, is this correct? (these match the data sheet, so i can pop them in my local branch, moving pca9555 support to a common location)

Reply:
Ah, okay.  I think it needs to be more explicit, to be honest, so it's not ... confusing, like this.

@jackhumbert @skullydazed your opinions?

Reply:
I'm sorry, looking at it now the file I copied came from `keyboards/xd96`. The file from `keyboards/xd84` doesn't have a copyright notice.

Though what I called bugs are precent in both. Maybe a bug is a bit of an overstatement, and if so I'm sorry.

What I'm talking about is the usage of the command enums.
In the original code `pca9555_set_config` uses the `CMD_OUTPUT_x` values and `pca9555_set_output` uses the `CMD_CONFIG_x` values. 
I'm a bit out of my comfort zone here, but what I get from reading the datasheet and just going by the names of the functions I decided to swap the usage so that `pca9555_set_config` uses the `CMD_CONFIG_x` values and vice versa.

Also seeing how the pca9555 now is in use by at least 3 boards. Maybe this could be moved to the `drivers` in some fashion? Just a thought though as I don't know what the consequences and long term plan regarding this.

Reply:
Good catch on the use of the enum values, i will have to correct the use elsewhere.

>Also seeing how the pca9555 now is in use by at least 3 boards. Maybe this could be moved to the drivers in some fashion

That is what i have locally, and eluded to when i mentioned "moving pca9555 support to a common location" :smile:. Given it can take a while for core changes, my preference would be this PR ships its copy of pca9555.[ch] and i refactor it in addition to the xd84 and xd96. That way it doesn't matter how long it takes to get reviews from the QMK members/collaborators .

Reply:
> That is what i have locally, and eluded to when i mentioned "moving pca9555 support to a common location" 😄.

I see, I didn't catch that. I guess great minds think alike 😸

> Given it can take a while for core changes, my preference would be this PR ships its copy of pca9555.[ch] and i refactor it in addition to the xd84 and xd96. That way it doesn't matter how long it takes to get reviews from the QMK members/collaborators .

This sounds like a good approach.